pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -606,20 +875,23 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 			// Resource is in use. The pod has to wait.
 			return nil, statusUnschedulable(logger, "resourceclaim in use", "pod", klog.KObj(pod), "resourceclaim", klog.KObj(claim))
 		}
-		if claim.Status.Allocation != nil &&
-			claim.Status.Allocation.AvailableOnNodes != nil {
-			nodeSelector, err := nodeaffinity.NewNodeSelector(claim.Status.Allocation.AvailableOnNodes)
-			if err != nil {
-				return nil, statusError(logger, err)
+
+		if claim.Status.Allocation != nil {
+			if claim.Status.Allocation.AvailableOnNodes != nil {
+				nodeSelector, err := nodeaffinity.NewNodeSelector(claim.Status.Allocation.AvailableOnNodes)
+				if err != nil {
+					return nil, statusError(logger, err)
+				}
+				s.informationsForClaim[index].availableOnNode = nodeSelector
 			}
-			s.informationsForClaim[index].availableOnNode = nodeSelector
-		}
-		if claim.Status.Allocation == nil &&
-			claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeWaitForFirstConsumer {
+
+			s.informationsForClaim[index].structuredParameters = slices.Contains(claim.Finalizers, resourcev1alpha2.Finalizer)
```

Comment:
What does this mean? shouldn't we already check the claims's ref class's `StructuredParameters` field?

Reply:
This is for an allocated claim. The class is no longer required to exist, so we have to detect "structured parameters" based on the claim attributes.

 I've added a comment:
>  The claim was allocated by the scheduler if it has the finalizer that is reserved for Kubernetes.